WABT is not supported on ARM64 windows

## Environment:
Host: Windows 10, x86_64
Target: Windows 11, ARM64.
Compiler:
Visual Studio 2022 Version 17.4.4

WABT Version:
```shell
git rev-parse HEAD
8a7b7497bdf78f9099f8d5a3a2c9bde87ddd52da
```

## Reproduce steps:
```shell
mkdir build_win_arm
cd build_win_arm
cmake -G "Visual Studio 17 2022" -A ARM64 ..
cmake --build .
```
### Observed behavior
Failed with error
```shell
 config.cc
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\config.cc) [D:\code\workspace\github\wabt\build_win_
arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\binary-writer.cc) [D:\code\workspace\github\wabt\bui
ld_win_arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\c-writer.cc) [D:\code\workspace\github\wabt\build_wi
n_arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\binary.cc) [D:\code\workspace\github\wabt\build_win_
arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\common.cc) [D:\code\workspace\github\wabt\build_win_
arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\apply-names.cc) [D:\code\workspace\github\wabt\build
_win_arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\binary-reader-ir.cc) [D:\code\workspace\github\wabt\
build_win_arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\binary-reader-logging.cc) [D:\code\workspace\github\
wabt\build_win_arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\color.cc) [D:\code\workspace\github\wabt\build_win_a
rm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\binary-reader.cc) [D:\code\workspace\github\wabt\bui
ld_win_arm\wabt.vcxproj]
D:\code\workspace\github\wabt\build_win_arm\include\wabt/config.h(175,1): fatal  error C1189: #error:  unexpected architecture (compiling source file D:\code\workspace\github\wabt\src\binary-writer-spec.cc) [D:\code\workspace\github\wab
t\build_win_arm\wabt.vcxproj]
```

### Expected Behavior
Build success

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,6 +113,11 @@ if (NOT USE_INTERNAL_SHA256)
 endif()
 
 if (WIN32)
+  if(${CMAKE_GENERATOR_PLATFORM} MATCHES "ARM64")
+    # https://discourse.cmake.org/t/visual-studio-error-unable-to-deploy-local-file-c-x64-release-zero-check/2072
+    # Target ZERO_CHECK blocks remote debugger of ARM64 in Visual Studio
+    set(CMAKE_SUPPRESS_REGENERATION ON)
+  endif()
   check_symbol_exists(ENABLE_VIRTUAL_TERMINAL_PROCESSING "windows.h" HAVE_WIN32_VT100)
 endif ()
 

--- a/include/wabt/interp/interp-math.h
+++ b/include/wabt/interp/interp-math.h
@@ -22,7 +22,7 @@
 #include <string>
 #include <type_traits>
 
-#if COMPILER_IS_MSVC
+#if COMPILER_IS_MSVC && _M_X64
 #include <emmintrin.h>
 #include <immintrin.h>
 #endif
@@ -189,7 +189,7 @@ RunResult WABT_VECTORCALL IntRem(T lhs, T rhs, T* out, std::string* out_msg) {
   return RunResult::Ok;
 }
 
-#if COMPILER_IS_MSVC
+#if COMPILER_IS_MSVC && _M_X64
 template <typename T> T WABT_VECTORCALL FloatAbs(T val);
 template <typename T> T WABT_VECTORCALL FloatCopysign(T lhs, T rhs);
 

--- a/src/config.h.in
+++ b/src/config.h.in
@@ -158,7 +158,7 @@ inline int Clz(unsigned int mask) {
 }
 
 inline int Clz(unsigned __int64 mask) {
-#if _M_X64
+#if _M_X64 || _M_ARM64
   if (mask == 0)
     return 64;
 
@@ -190,7 +190,7 @@ inline int Ctz(unsigned int mask) {
 }
 
 inline int Ctz(unsigned __int64 mask) {
-#if _M_X64
+#if _M_X64 || _M_ARM64
   if (mask == 0)
     return 64;
 
@@ -208,12 +208,31 @@ inline int Ctz(unsigned __int64 mask) {
 #endif
 }
 
-inline int Popcount(uint8_t value) {
-  return __popcnt(value);
+#if _M_ARM64
+//https://stackoverflow.com/a/70012905
+template <typename T>
+int BrianKernighanPopcount(T value) {
+  int count;
+  for(count = 0; value; count++)
+  {
+    value &= value - 1;
+  }
+  return count;
 }
+#endif
 
 inline int Popcount(unsigned long value) {
+  #if _M_X64 || _M_IX86
   return __popcnt(value);
+#elif _M_ARM64
+  return BrianKernighanPopcount(value);
+#else
+#error unexpected architecture
+#endif
+}
+
+inline int Popcount(uint8_t value) {
+  return Popcount((unsigned long)value);
 }
 
 inline int Popcount(unsigned int value) {
@@ -225,6 +244,8 @@ inline int Popcount(unsigned __int64 value) {
   return __popcnt64(value);
 #elif _M_IX86
   return Popcount(HighDword(value)) + Popcount(LowDword(value));
+#elif _M_ARM64
+  return BrianKernighanPopcount(value);
 #else
 #error unexpected architecture
 #endif

